apps/prairielearn/src/lib/question-testing.js

```diff
@@ -40,21 +41,15 @@ function createTestSubmission(
 ) {
   debug('_createTestSubmission()');
   if (question.type !== 'Freeform') return callback(new Error('question.type must be Freeform'));
-  let questionModule, question_course, courseIssues, data, submission_id, grading_job;
+  const questionModule = questionServers.getModule(question.type);
+  let question_course, courseIssues, data, submission_id, grading_job;
   async.series(
     [
-      (callback) => {
-        questionServers.getModule(question.type, (err, ret_questionModule) => {
-          if (ERR(err, callback)) return;
-          questionModule = ret_questionModule;
-          debug('_createTestSubmission()', 'loaded questionModule');
-          callback(null);
-        });
-      },
       async () => {
         question_course = await getQuestionCourse(question, variant_course);
       },
       (callback) => {
+        assert(questionModule.test, `Question type ${question.type} does not support testing.`);
```

Comment:
Will `async.series` correctly capture a thrown error and propagate it? This is essentially the same problem as the other comments I've been leaving, but it's possible that `async.series` specifically would handle this case.

Reply:
This code only runs when the type is Freeform, so test should exist. I added the abort so the typechecker wouldn't complain. I changed it now to only run if not null.